### PR TITLE
Extend API documentation

### PR DESCRIPTION
**Pull request by stes**
- Remove the duplicate "optional" classifiers currenlty in the text, this is done automatically by sphinx
- Remove the "temporary" note from `api.rst`
- Add more verbose descriptions to the task

### CODE REVIEW

**Comment by stes** [comment]
Ok there's some overlap with #28 , but the other changes still apply. Also, reordered the docs in ``api.rst`` to make the tasks appear first, which is more relevant for users. I excluded the torchvision models for now, as these do not contain any public classes yet.

**Comment by zimmerrol** [comment]
Let's first merge #28 before reviewing and merging this one.

File: shifthappens/tasks/base.py [hunk 1]
```diff
@@ -151,42 +176,79 @@ def evaluate(self, model: sh_models.Model) -> Optional[TaskResult]:
...
 
     pass
 
 
 class OODScoreTaskMixin:
-    """Indicates that the task requires the model to return the OOD scores."""
+    """Indicates that the task requires the model to return the OOD scores.
+
+    Tasks implementing this mixin will be provided with the ``ood_scores`` attribute in the
+    ``shifthappens.models.base.ModelResult`` returned during evaluation.
```
Comment (zimmerrol) [inline shifthappens/tasks/base.py:241]: This should be changed to something like:

    :py:class:`shifthappens.models.base.ModelResult`

to make sure sphinx uses the correct links to the respective class documentations.

The same also applies to tall other occurences.

**Review by zimmerrol — CHANGES_REQUESTED** [review]

File: shifthappens/tasks/base.py (hunk 1 above)
Comment (stes) [reply shifthappens/tasks/base.py]: done (hope I got all of them)

**Comment by stes** [comment]
@zimmerrol cross refs are fixed, hope I didnt miss any

**Comment by stes** [comment]
Also, this is rebased onto #28 now, I'll change base branch for a better diff. Please merge #28 first, then merge this one.

**Comment by zimmerrol** [comment]
Once you've reviewed #28 I can merge both PRs.

File: shifthappens/tasks/base.py [hunk 2]
```diff
@@ -151,42 +176,79 @@ def evaluate(self, model: sh_models.Model) -> Optional[TaskResult]:
...
+        """
         raise NotImplementedError()
 
     @abstractmethod
     def _evaluate(self, model: sh_models.Model) -> TaskResult:
-        """Implement this function to evaluate the task and return a dictionary with the
-        calculated metrics."""
+        """Evaluate the task and return a dictionary with the calculated metrics.
+
+        model: The passed model implementents a ``predict`` function returning an iterator
```
Comment (zimmerrol) [inline shifthappens/tasks/base.py:195]: Let's also use the same docstring style as for the rest, i.e.:

    Args:
        model: The passed model ...

**Comment by stes** [comment]
> We could improve one small thing regarding docstring-styles about this PR before merging it.

we'll use [google style doc strings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html), right?

**Comment by zimmerrol** [comment]
Yep, that is correct. I think only at one location (see my comment above) you used a different style.

**Review by zimmerrol — APPROVED** [review]
LGTM

**Pull request merged**